# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/Stencil.cpp
# Commit: 73cc5fa0e0bd
# Full Hash: 73cc5fa0e0bdd7a1d6b0f8450b1541c4bc53c333
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 9: Use TaggedParserAtomIndex in RegExpStencil.atom_. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96916
# ==============================================================================

diff -r f505b8fb8f8b -r 73cc5fa0e0bd js/src/frontend/Stencil.cpp
--- a/js/src/frontend/Stencil.cpp	Sat Nov 14 00:03:07 2020 +0000
+++ b/js/src/frontend/Stencil.cpp	Sat Nov 14 00:03:20 2020 +0000
@@ -1008,15 +1008,23 @@
 void RegExpStencil::dump() {
   js::Fprinter out(stderr);
   js::JSONPrinter json(out);
-  dump(json);
+  dump(json, nullptr);
 }
 
-void RegExpStencil::dump(js::JSONPrinter& json) {
-  GenericPrinter& out = json.beginString();
+void RegExpStencil::dump(js::JSONPrinter& json,
+                         CompilationStencil* compilationStencil) {
+  json.beginObject();
+  dumpFields(json, compilationStencil);
+  json.endObject();
+}
 
-  out.put("/");
-  atom_->dumpCharsNoQuote(out);
-  out.put("/");
+void RegExpStencil::dumpFields(js::JSONPrinter& json,
+                               CompilationStencil* compilationStencil) {
+  json.beginObjectProperty("pattern");
+  DumpTaggedParserAtomIndex(json, atom_, compilationStencil);
+  json.endObject();
+
+  GenericPrinter& out = json.beginStringProperty("flags");
 
   if (flags_.global()) {
     out.put("g");
@@ -1037,7 +1045,7 @@
     out.put("y");
   }
 
-  json.endString();
+  json.endStringProperty();
 }
 
 void BigIntStencil::dump() {
@@ -1619,7 +1627,7 @@
 
   json.beginListProperty("regExpData");
   for (auto& data : regExpData) {
-    data.dump(json);
+    data.dump(json, this);
   }
   json.endList();
 